From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Justin Hammond <Justin@dynam.ac>
Date: Mon, 19 Oct 2026 20:11:06 +0800
Subject: [PATCH] add debug mode

Port of the LicheeRV Nano debug mode patch from plat/sg200x to plat/cv181x.

diff --git a/plat/cv181x/bl2/bl2_opt.c b/plat/cv181x/bl2/bl2_opt.c
index ccf55a2..5a341c2 100644
--- a/plat/cv181x/bl2/bl2_opt.c
+++ b/plat/cv181x/bl2/bl2_opt.c
@@ -566,6 +566,20 @@ retry_from_flash:
 	monitor_entry = run_addr;
 #endif
 
+	if (!(mmio_read_32(GPIO_BASE + 0x050) & (1 << 30))) {
+		NOTICE("=========================================\n");
+		NOTICE("||             Debug Mode              ||\n");
+		NOTICE("||                                     ||\n");
+		NOTICE("=========================================\n");
+
+		mmio_setbits_32(0x3003024, 1 << 6);	/* reset the small core */
+
+		/* pinmux was set for jtag by default */
+
+		while(1)
+			;
+	}
+
 	if (monitor_entry) {
 		NOTICE("Jump to monitor at 0x%lx.\n", monitor_entry);
 		jump_to_monitor(monitor_entry, loader_2nd_entry);
-- 
-

